src/group_theory/submonoid/pointwise.lean

```diff
@@ -141,6 +143,21 @@ instance pointwise_central_scalar [mul_distrib_mul_action αᵐᵒᵖ M] [is_cen
   is_central_scalar α (submonoid M) :=
 ⟨λ a S, congr_arg (λ f, S.map f) $ monoid_hom.ext $ by exact op_smul_eq_smul _⟩
 
+/-- The submonoid of powers of a natural number of a commutative submonoid.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def pointwise_has_pow {M : Type*} [comm_monoid M] : has_pow (submonoid M) ℕ :=
+⟨λ S n, S.map $ pow_monoid_hom n⟩
+
+/-- The submonoid of powers of an integer of a commutative subgroup.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def pointwise_has_zpow {G : Type*} [comm_group G] : has_pow (submonoid G) ℤ :=
+⟨λ S n, S.map $ zpow_group_hom n⟩
+
+localized "attribute [instance] submonoid.pointwise_has_pow submonoid.pointwise_has_zpow"
+  in pointwise
```

Comment:
I think these probably don't belong in this `monoid` section, since neither of them are actually talking about `monoid`. I'd lean towards putting them in a new section on line 115.